File: src/main/java/ru/ioleksiv/telegram/bot/api/model/method/sticker/DeleteStickerFromSet.java
package ru.ioleksiv.telegram.bot.api.model.method.sticker;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.api.model.NetworkError;
import ru.ioleksiv.telegram.bot.core.controller.network.ActionNetworker;
import ru.ioleksiv.telegram.bot.core.model.method.RunnableAction;
import ru.ioleksiv.telegram.bot.core.model.responses.BooleanResponse;
import ru.ioleksiv.telegram.bot.core.model.responses.CommonResponse;

/**
 * @see <a href="https://core.telegram.org/bots/api#deletestickerfromset>deleteStickerFromSet</a>
 */
public class DeleteStickerFromSet extends RunnableAction<Boolean> {
    private static final String METHOD = "deleteStickerFromSet";
    /**
     * sticker String Yes File identifier of the sticker
     */
    @JsonProperty("sticker")
    private String sticker = null;

    public DeleteStickerFromSet(ActionNetworker actionNetworker) {
        super(METHOD, actionNetworker);
    }

    public String getSticker() {
        return sticker;
    }

    public DeleteStickerFromSet setSticker(String sticker) {
        this.sticker = sticker;
        return this;
    }

    @Override
    protected Class<? extends CommonResponse<Boolean>> getResultWrapperClass() {
        return BooleanResponse.class;
    }

    @Override
    public DeleteStickerFromSet setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/sticker/DeleteStickerFromSet.kt
package io.github.oleksivio.tl.kbot.server.api.method.sticker

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.BooleanResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.Action

/**
 * @see [deleteStickerFromSet](https://core.telegram.org/bots/api/#deletestickerfromset)
 */
class DeleteStickerFromSet(
    /**
     * sticker String Yes File identifier of the sticker
     */
    @JsonProperty(ApiDict.STICKER_KEY)
    val sticker: String
) : Action<Boolean>() {

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "deleteStickerFromSet"

    @JsonIgnore
    override val resultWrapperClass = BooleanResponse::class
}


